code/game/turfs/turf.dm

```diff
@@ -408,8 +408,11 @@
 /turf/proc/clean(floor_only)
 	for(var/obj/effect/decal/cleanable/C in src)
 		var/obj/effect/decal/cleanable/blood/B = C
+		var/obj/effect/decal/cleanable/tar/T = C
 		if(istype(B) && B.off_floor)
 			floor_only = FALSE
+		else if(istype(T)) // We need to remove the slowdown from tar instead of just qdeling
+			T.remove_tar()
```

Comment:
Instead of doing this, put it in the tar's Destroy() method, which is called when the tar is qdel'd. This will ensure that the slowdown is properly removed even if the tar is removed by some other means

Reply:
Knowing about Destroy() would have saved me so much time. At least it's here now.